src/arena.c

```diff
@@ -1259,13 +1262,21 @@ static size_t
 arena_prof_demote(tsdn_t *tsdn, edata_t *edata, const void *ptr) {
 	cassert(config_prof);
 	assert(ptr != NULL);
+	size_t usize = isalloc(tsdn, ptr);
+	size_t bumped_usize = PAGE_CEILING(usize);
+	/*
+	 * TODO(Svetlitski) This doesn't properly handle the case
+	 * where the allocation was explicitly over-aligned.
+	 */
```

Comment:
With regards to this comment, I'm not really sure there's a way to solve this. If the allocation was over-aligned to an alignment that is greater than `PAGE` (potentially causing the correct value of `bumped_usize` to actually be greater than`PAGE_CEILING(usize)`), I don't see how we can possibly recover that information.

Reply:
I suppose we could just skip sampling allocations with a requested alignment of greater than `PAGE` in the first place.

Reply:
> if the allocation was over-aligned to an alignment that is greater than PAGE

This should result the usize being a large size, if we use sa2u to set the bumped_usize, i.e. alignment >4k cannot be a small size class. Does that solve this problem?

Not sampling these over aligned allocations isn't really a good option -- imagine if there's a problematic function doing aligned malloc but with an unnecessarily large alignment, and it's cause high memory usage, then heap profiles wouldn't cover that part.

The only case we current would skip sampling or other periodic events, is when we are in a reentrancy situation, i.e. malloc => some user hook => malloc, then all the non essential features are bypassed in that nested malloc. This is rather a last resort to avoid circular dependencies (e.g. the first malloc might be sampled and is invoking the one of the sampling hooks).

Reply:
That makes sense. Before one of your previous comments I hadn't realized `usize` had already been adjusted to reflect the user-requested alignment requirement by the time we get to `imalloc_sample` and the like.